### PR TITLE
Add support to SuperPMI for running with protononjit.dll on x86

### PR DESCRIPTION
**Pull request by echesakovMSFT**
For some reason only ARM64 AltJit was supported

### CODE REVIEW

File: src/ToolBox/superpmi/superpmi/commandline.cpp [hunk 1]
```diff
@@ -593,11 +593,28 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
...
-        LogError("Illegal target architecture specified with -target (only arm64 is supported).");
-        DumpHelp(argv[0]);
-        return false;
+        const char* errorMessage = nullptr;
+#ifdef _TARGET_AMD64_
+        if (0 != _stricmp(o->targetArchitecture, "arm64"))
+        {
+            errorMessage = "Illegal target architecture specified with -target (only arm64 is supported).";
+        }
+#elif _TARGET_X86_
```
Comment (BruceForstall) [inline src/ToolBox/superpmi/superpmi/commandline.cpp]: => `#elif defined(_TARGET_X86_)`

File: src/ToolBox/superpmi/superpmi/jitinstance.cpp [hunk 2]
```diff
@@ -314,30 +314,40 @@ JitInstance::Result JitInstance::CompileMethod(MethodContext* MethodToCompile, i
...
+        if (jitResult == CORJIT_SKIPPED)
         {
             // For altjit, treat SKIPPED as OK
-            temp = CORJIT_OK;
+#ifdef _TARGET_AMD64_
+            if (SpmiTargetArchitecture == SPMI_TARGET_ARCHITECTURE_ARM64)
+            {
+                jitResult = CORJIT_OK;
+            }
+#elif _TARGET_X86_
```
Comment (BruceForstall) [inline src/ToolBox/superpmi/superpmi/jitinstance.cpp]: => `#elif defined(_TARGET_X86_)`

File: src/ToolBox/superpmi/superpmi/commandline.cpp (hunk 1 above)
Comment (echesakovMSFT) [reply src/ToolBox/superpmi/superpmi/commandline.cpp]: Good catch, thanks

File: src/ToolBox/superpmi/superpmi/commandline.cpp [hunk 3]
```diff
@@ -593,11 +593,28 @@ bool CommandLine::Parse(int argc, char* argv[], /* OUT */ Options* o)
...
+    if (o->targetArchitecture != nullptr)
     {
-        LogError("Illegal target architecture specified with -target (only arm64 is supported).");
-        DumpHelp(argv[0]);
-        return false;
+        const char* errorMessage = nullptr;
+#ifdef _TARGET_AMD64_
+        if (0 != _stricmp(o->targetArchitecture, "arm64"))
+        {
+            errorMessage = "Illegal target architecture specified with -target (only arm64 is supported).";
```
Comment (sandreenko) [inline src/ToolBox/superpmi/superpmi/commandline.cpp]: I think we can add "only arm64 is supported on x64 host".

File: src/ToolBox/superpmi/superpmi/superpmi.cpp [hunk 4]
```diff
@@ -45,9 +45,20 @@ void SetSuperPmiTargetArchitecture(const char* targetArchitecture)
...
+    else
+    {
+        SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_X86;
+    }
 #elif defined(_TARGET_ARM_)
     SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_ARM;
+#elif defined(_TARGET_ARM64)
+    SpmiTargetArchitecture = SPMI_TARGET_ARCHITECTURE_ARM64
+#else
+#error Unsupported target architecture
```
Comment (sandreenko) [inline src/ToolBox/superpmi/superpmi/superpmi.cpp]: Is it unsupported target or host architecture?

Comment (echesakovMSFT) [reply src/ToolBox/superpmi/superpmi/superpmi.cpp]: In this particular case it's both 🤣 I will change the error to be just "Unsupported architecture"

**Pull request merged**